components/tools/OmeroWeb/omeroweb/webclient/static/webclient/javascript/ome.webclient.actions.js

```diff
@@ -382,9 +381,13 @@ OME.refreshThumbnails = function(options) {
     var $thumbs = $(spw_selector + ", " + search_selector);
     if ($thumbs.length > 0){
         $thumbs.each(function(){
-            var $this = $(this),
-                base_src = $this.attr('src').split('?')[0];
-            $this.attr('src', base_src + "?_="+rdm);
+            var _t = $(this);
+                OME.load_thumbnail(
+                    options.imageId, options.thumbnail_url,
+                    function(thumb) {
+                        _t.attr('src', thumb);
+                    }
+                );
```

Comment:
As discussed, these thumbnails can be refreshed in a batch instead of 1 call per thumbnail.